src/azul/__init__.py

```diff
@@ -546,7 +543,15 @@ def from_json(cls, name: str, spec: JSON) -> 'Config.Catalog':
             return cls(name=name,
                        atlas=spec['atlas'],
                        internal=spec['internal'],
-                       plugins=plugins)
+                       plugins=plugins,
+                       sources=set(spec['sources']))
+
+        @classmethod
+        def validate_name(cls, catalog, **kwargs):
+            reject(cls._catalog_name_re.fullmatch(catalog) is None,
+                   'Catalog name contains invalid characters',
+                   catalog,
+                   **kwargs)
```

Comment:
https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#14line-wrapping-and-indentation